File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
import java.util.PriorityQueue;

class Solution {
    public int solution(int left, int right) {
        int answer =0;

        int count = 0;
        for(int i=left; i<=right; i++){
            count = math(i);

            if (count%2==0) answer += i;
            else answer-= i;
        }
        return answer;
    }

    public int math(int num){
        int n=0;
        for(int i=1; i<=num; i++) {
            if (num%i==0) n++;
        }
        return n;
    }
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.swift
import Foundation

func solution(_ left:Int, _ right:Int) -> Int {
    var sum = 0
    for i in left...right { 
        var length = getDivisors(of: i)
        if length.count % 2 == 0 {
            sum += i
        } else { 
            sum -= i
        }
    }
    
    return sum
}

func getDivisors(of number: Int) -> [Int] {
    var divisors: [Int] = []
    for i in 1...number {
        if number % i == 0 {
            divisors.append(i)
        }
    }
    return divisors
}